The NullPointerException will never be thrown as all exceptions will be caught by the first catch block.